src/main/kotlin/no/nav/familie/tilbake/dokumentbestilling/manuell/brevmottaker/ManuellBrevmottakerService.kt

```diff
@@ -0,0 +1,79 @@
+package no.nav.familie.tilbake.dokumentbestilling.manuell.brevmottaker
+
+import no.nav.familie.kontrakter.felles.historikkinnslag.Aktør
+import no.nav.familie.tilbake.api.dto.ManuellBrevmottakerDto
+import no.nav.familie.tilbake.common.exceptionhandler.Feil
+import no.nav.familie.tilbake.historikkinnslag.HistorikkService
+import no.nav.familie.tilbake.historikkinnslag.TilbakekrevingHistorikkinnslagstype
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDateTime
+import java.util.UUID
+
+@Service
+class ManuellBrevmottakerService(
+    private val manuellBrevmottakerRepository: ManuellBrevmottakerRepository,
+    private val historikkService: HistorikkService
+) {
+
+    @Transactional
+    fun leggTilBrevmottaker(behandlingId: UUID, manuellBrevmottakerDto: ManuellBrevmottakerDto) {
+        val manuellBrevmottaker = ManuellBrevmottakerMapper.tilDomene(behandlingId, manuellBrevmottakerDto)
+        manuellBrevmottakerRepository.insert(manuellBrevmottaker)
+        historikkService.lagHistorikkinnslag(
+            behandlingId = behandlingId,
+            historikkinnslagstype = TilbakekrevingHistorikkinnslagstype.BREVMOTTAKER_LAGT_TIL,
+            aktør = Aktør.SAKSBEHANDLER,
+            opprettetTidspunkt = LocalDateTime.now()
+        )
+    }
+
+    fun hentBrevmottakere(behandlingId: UUID) =
+        manuellBrevmottakerRepository.findByBehandlingId(behandlingId).map { ManuellBrevmottakerMapper.tilRespons(it) }
```

Comment:
Hmm, er det kanskje bedre å gjøre mappingen til DTO i kontrolleren?

Jeg syntes at det er litt rart at man kan få en Brevmottaker med nullable id ut av denne metoden.
Kommer de fra databasen er det jo garantert at man har en id.

Alternativet er å kanskje splitte opp slik at man har to forskjellige dto (en for request og en for respons, der respons har non-nullable id).

Reply:
Du har et poeng, men det er tilsvarende mønster i ba-sak, så tenker kanskje det får være greit her også :)